src/main/java/ua/com/juja/microservices/teams/slackbot/service/impl/TeamServiceImpl.java

```diff
@@ -41,27 +41,51 @@ public TeamServiceImpl(TeamRepository teamRepository, UserService userService) {
     }
 
     @Override
-    public Team activateTeam(String text) {
+    public Team activateTeam(String fromUser, String text) {
         Utils.checkNull(text, "Text must not be null!");
-        List<String> slackNames = SlackNameHandler.getSlackNamesFromText(text);
+        Utils.checkNull(fromUser, "FromUser must not be null!");
+        Set<String> slackNames = SlackNameHandler.getSlackNamesFromText(text);
         if (slackNames.size() != TEAM_SIZE) {
             throw new WrongCommandFormatException(String.format("We found %d slack names in your command." +
                     " But size of the team must be %s.", slackNames.size(), TEAM_SIZE));
         }
-        Set<User> users = new HashSet<>(userService.findUsersBySlackNames(slackNames));
-        if (users.size() != slackNames.size()) {
-            throwUserExchangeException();
-        }
-        Set<String> uuids = users.stream()
-                .map(User::getUuid)
-                .collect(Collectors.toSet());
-        TeamRequest teamRequest = new TeamRequest(uuids);
-        Team activatedTeam = teamRepository.activateTeam(teamRequest);
-        checkTeamMembersEquality(teamRequest.getMembers(), activatedTeam.getMembers());
+        fromUser = SlackNameHandler.addAtToSlackName(fromUser);
+        slackNames.add(fromUser);
+        Set<User> users = new HashSet<>(userService.findUsersBySlackNames(new ArrayList<>(slackNames)));
+        String fromUserUuid = getFromUserUuid(fromUser, users);
+        Set<String> membersUuids = getMembersUuids(fromUser, users, TEAM_SIZE);
+        ActivateTeamRequest activateTeamRequest = new ActivateTeamRequest(fromUserUuid, membersUuids);
+        Team activatedTeam = teamRepository.activateTeam(activateTeamRequest);
+        checkTeamMembersEquality(activateTeamRequest.getMembers(), activatedTeam.getMembers());
         log.info("Team activated: '{}'", activatedTeam.getId());
         return activatedTeam;
     }
 
+    private String getFromUserUuid(String fromUserSlackName, Set<User> users) {
+        log.debug("Before extract fromUser from users : '{}'.FromUser is '{}'", users, fromUserSlackName);
+        User fromUser = users.stream()
+                .filter(user -> user.getSlack().equals(fromUserSlackName))
+                .collect(Collectors.toList()).get(0);
+        log.debug("After extract fromUser from users map. Uuid is '{}'", fromUser.getUuid());
+        return fromUser.getUuid();
+    }
+
+    private Set<String> getMembersUuids(String fromUser, Set<User> users, int expectedSize) {
+        log.debug("Before extract members from users : '{}'. Expected size '{}'. FromUser is '{}'", users,
+                expectedSize, fromUser);
+        Set<String> uuids;
+        if (users.size() == expectedSize) {
```

Comment:
I'd better add a comments for these if-else to understand business logic. E.g. `if .... //if a fromUser is a part of the team` and `else .... //if a fromUser is not a part of the team`.
Because I spent some time to understand why you wrote such way.